### PR TITLE
Authentication Monitoring Changes and Module Refactor

### PR DESCRIPTION
**Pull request by ehatch90**
Added monitoring to the caching layer to log events and publish metrics around authentication traffic. To more easily support this, factored authentication out to its own module and service.

### CODE REVIEW

File: src/main/java/org/kairosdb/auth/PulseAuthModule.java [hunk 1]
```diff
@@ -0,0 +1,785 @@
...
+        public List<String> doFetchRoles() throws Exception
+        {
+            return getUserRoles(_rootContext, getUserName(), attributes);
+        }
+
+    }
+
+
+    /**
+     * get the available information about the user
```
Comment (mkeagar) [inline src/main/java/org/kairosdb/auth/PulseAuthModule.java]: In general I think you should use consistent formatting for your comments. Sometimes you are using capitalization for sentences and sometimes you aren't. I suggest changing all comments to use correct capitalization.

File: src/main/resources/kairosdb.conf [hunk 2]
```diff
@@ -59,21 +59,42 @@ kairosdb: {
...
+    # Samples of how to configure the various auth versions can be found here https://www.eclipse.org/jetty/documentation/current/jaas-support.html
+    # ldap-loginModule.conf will configure the ldap module.
+    # the auth_module_name below should match the name outside the curly braces in the respective conf file.
+    # The basics of how to configure two or more modules to work together in logic and/or combinations can be found here https://docs.oracle.com/javase/8/docs/api/javax/security/auth/login/Configuration.html and here https://github.com/rundeck/rundeck/wiki/Multiple-authentication-modules
+    # Service.jetty_auth as well as module_name in jetty_auth section must be configured and uncommented for auth to be enabled.
+    #module_name: "cachingLDAPAuth"
+
+    # To enable softAuthentication(allows no credentials, so services can enable auth over time)LDAPAuth functionality set the following property to true.
+    # Specifically enables the org.kairosdb.auth.PulseDefaultAuthenticatiorFactory and org.kairosdb.auth.PulseBasicAuthenticator to be used in conjunction with the org.kairosdb.auth.PulseAuthModulee.
```
Comment (guildencrantz) [inline src/main/resources/kairosdb.conf]: s/PulseAuthModulee/PulseAuthModule/

File: src/main/resources/kairosdb.conf [hunk 3]
```diff
@@ -59,21 +59,42 @@ kairosdb: {
...
+    # also requires setting the appropriate configuration for the LDAPLoginModule in the ldap-loginModule.conf file above including the file property for the PropertyFile LoginModule.
+    #use_soft_authentication: false
+
+    # The amount of time, in minutes, a user will remain in the cache after logging in
+    #user_cache_duration: 60
+
+    # The amount of time, in minutes, a user + password combination will be locked out of the system from failed logins
```
Comment (guildencrantz) [inline src/main/resources/kairosdb.conf:94]: From a progression perspective these two `lockout` variables should be swapped (so as
you read through the example you see `lockout_cache_threshold` first, which
tells you how many failed attempts get you locked, _then_
`lockout_cache_duration` tells you how long to wait before _unlocking_ that
user.)

**Review by guildencrantz — APPROVED** [review]

File: src/main/java/org/kairosdb/auth/PulseAuthModule.java (hunk 1 above)
Comment (ehatch90) [reply src/main/java/org/kairosdb/auth/PulseAuthModule.java]: Fixed on latest commit

File: src/main/resources/kairosdb.conf (hunk 2 above)
Comment (ehatch90) [reply src/main/resources/kairosdb.conf]: Fixed on latest commit

File: src/main/resources/kairosdb.conf (hunk 3 above)
Comment (ehatch90) [reply src/main/resources/kairosdb.conf:94]: Fixed on latest commit

**Review by mkeagar — APPROVED** [review]
Approving after reviewing with Eric.

**Pull request merged**